src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -124,20 +138,27 @@ function VisEditor($scope, $route, timefilter, AppState, $location, kbnUrl, $tim
     vis: savedVisState
   };
 
+  // Instance of app_state.js.
   let $state = $scope.$state = (function initState() {
-    $state = new AppState(stateDefaults);
 
-    if (!angular.equals($state.vis, savedVisState)) {
+    // This is used to generate a 'uiState' PersistedState instance, and
+    // implictly add 'change' and 'fetch_with_changes' event handlers.
+    const appState = new AppState(stateDefaults);
```

Comment:
I don't see how the `change` or `fetch_with_changes` listeners are of any importance here. All that maters is that the properties of the appState are synced with the url, right? In order to save changes call `appState.save()` any changed from the url will be reflected in the properties of the `appState`